### DISCUSSION TITLE
Can I get User Id in Edge Function?

### DISCUSSION THREAD

**Discussion by sarkarology** [supabase/supabase#8371] [Questions] [answered]
On the browser side we get user object using: supabaseClient.auth.user()

But there's no such function on the Edge Function as far as I am trying.

The official example works fine: https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts

But on using supabaseClient.auth.session() on edge function, I can see the token is the ANON key, and the user property of the session object is null. But how can I get the userId of the user that called that function?

Context: I need to insert User Id into a table column from Edge Function.

**Comment by GaryAustin1** [comment]
This Supabase example shows using the sent JWT to set the user into the Edge SB client.  
https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts
According to the docs setAuth also will return the user (although technically you don't need it if you use auth.uid() as the default for your column).
const { user, error } = supabase.auth.setAuth(access_token)

**Reply by sarkarology** [reply]
In my edge function, I tried:

const { user } = supabaseClient.auth.setAuth(req.headers.get('Authorization')!.replace('Bearer ', ''))

user returned is null.

**Reply by qnsi** [reply]
For googlers, make sure you are using real user token. I had similar problem as sarkalogy. My code in edge function was 

```
    const supabase = createClient(
      Deno.env.get("SUPABASE_URL") ?? "",
      Deno.env.get("SUPABASE_SERVICE_ROLE_KEY") ?? "",
    );
    const token = req.headers.get("Authorization").replace("Bearer ", "");
    const {
      data: { user },
    } = await supabase.auth.getUser(token);
```

similar to the example linked in parent, and I would get user as null. The problem is, I was using curl with a auth token that was not connected to any user. When I triggered function from my react native app, the correct user is getting retrieved

**Comment by sarkarology** [comment, marked as answer]
Found the solution. I can get the user using getUse()r api call from within edge function:

const { user: userData, error: userError } = await supabaseClient.auth.api.getUser(req.headers.get('Authorization')!.replace('Bearer ', ''))

Please let me know if there's any caveat with this approach.
